sophia/src/triple/stream/_error.rs

```diff
@@ -0,0 +1,214 @@
+use std::convert::Infallible;
+use std::error::Error;
+use std::fmt;
+
+/// A error that is raised by functions that move fallible `Source`s into
+/// fallible `Sinks`.
+///
+/// In case this error is raised it can be matched to investigate if `Source`
+/// or `Sink` failed.
+///
+/// The `std::error::Error::source()` method is implemented accordingly.
+///
+/// # ToDo
+///
+/// For a proper use in multi-threaded environments the trait bounds should be
+/// supplied with `+ Send + Sync`. However, such bounds wound make `StreamError`
+/// incompatible with `sophia`'s whole error-handling. This should be resolved
+/// when the error-handling is completly refactored
+/// ([tracking issue](https://github.com/pchampin/sophia_rs/issues/8)).
+#[derive(Debug)]
```

Comment:
Why not use `thiserror` already? This is the plan eventually, and this would reduce the number of lines of code...